pkg/identityserver/user_registry.go

```diff
@@ -432,7 +431,11 @@ func (is *IdentityServer) updateUser(ctx context.Context, req *ttnpb.UpdateUserR
 
 	if !updatedByAdmin {
 		req.User.PrimaryEmailAddressValidatedAt = nil
-		cleanContactInfo(req.User.ContactInfo)
+		req.FieldMask.Paths = ttnpb.AddFields(req.FieldMask.GetPaths(), "primary_email_address_validated_at")
+		if len(req.User.ContactInfo) > 0 {
+			cleanContactInfo(req.User.ContactInfo)
+			req.FieldMask.Paths = ttnpb.AddFields(req.FieldMask.GetPaths(), "contact_info")
```

Comment:
I would prefer that we don't add new fields to the field mask in this manner. This makes the `contact_info` field mask implicit if the value is not empty and the user is not an admin, which is not how field masks are supposed to work.

Reply:
While I agree with the sentiment and the conditional of `contact_info` not being an empty slice is unnecessary I do wonder if it is possible to enforce non admin behaviors without affecting the field mask minimally. 

For example, this feels acceptable to me
```go
req.FieldMask.Paths = ttnpb.AddFields(req.FieldMask.GetPaths(), "primary_email_address_validated_at")
```
but indeed the 
```go
if len(req.User.ContactInfo) > 0 {
	cleanContactInfo(req.User.ContactInfo)
	req.FieldMask.Paths = ttnpb.AddFields(req.FieldMask.GetPaths(), "contact_info")
```
is unecessary.

Reply:
The validate at timestamp is indeed a fine addition because we provide the value intentionally - we control the value put into the request on the RPC side, so the intent is clear.

The `contact_info` here is added just because the slice is non empty in the request, even though that slice comes from the caller - that is what I find wrong here. Caller input must be provided next to a field mask entry, otherwise it must always be ignored.